product_contract/migrations/12.0.6.0.0/pre-migration.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 ACSONE SA/NV.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    """Copy the product template contract on their variants."""
+    contract_key = "property_contract_template_id"  # TODO: check that mig (do in SQL?)
+
+    for company in env["res.company"].with_context(active_test=False).search([]):
```

Comment:
IMO we should avoid the use of the ORM.... but  The query could be a little bit tricky to write for company dependent fields